### PR TITLE
T1464

### PR DESCRIPTION
**Pull request by dseif**
...corn scripts. Made sure we snapshot the page before any of our popcorn scripts load

### CODE REVIEW

File: src/core/page.js [hunk 1]
```diff
@@ -52,24 +52,31 @@ define( [ "core/logger", "core/eventmanager" ], function( Logger, EventManagerWr
...
+        head = document.getElementsByTagName( "head" )[ 0 ].cloneNode( true );
+      } else {
         body = _snapshot.body.cloneNode( true );
+        head = _snapshot.head.cloneNode( true );
       }
 
-      head = document.getElementsByTagName( "head" )[ 0 ].cloneNode( true );
-
       toExclude = Array.prototype.slice.call( head.querySelectorAll( "*[data-butter-exclude]" ) );
+      toInclude = Array.prototype.slice.call( document.head.querySelectorAll( "*[data-butter-include]" ) );
```
Comment (cadecairos) [inline src/core/page.js:71]: `toInclude` is leaking to the global scope

Comment (dseif) [inline src/core/page.js:71]: ah, my bad

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/9b5106da2bcc28f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/9b5106da2bcc28f/output.txt

Total script time: 0.63 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3ee2d6d05a4a2b8/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/3ee2d6d05a4a2b8/output.txt

Total script time: 0.68 mins

Server is running on http://pr466.butyr.org/templates/test.html

File: src/core/page.js (hunk 1 above)
Comment (cadecairos) [inline src/core/page.js:71]: `document.head.querySelectorAll( "*[data-butter-include]" )` => `head.querySelectorAll( "*[data-butter-include]" )`

Comment (dseif) [inline src/core/page.js:71]: nope we want document.head, as head is a snapshotted clone from earlier. We want all of the scripts with data-butter-include that weren't snapshotted.

Comment (dseif) [inline src/core/page.js:71]: Yea I don't know about the line above, let me do some testing and report back

Comment (dseif) [inline src/core/page.js:71]: Yea, `head` should have basically everything except for the popcorn scripts at this point ( if it was snapshotted ). If it wasn't snapshotted then everything just goes along as normal, so this is fine.

**Comment by jbuck** [comment]
wont-fixed in Lighthouse

**Pull request closed without merging**